docs/real_world_fl/flare_api.rst

```diff
@@ -73,6 +77,84 @@ can be run::
 
 Monitor Job
 ^^^^^^^^^^^
-By default, ``monitor_job()`` waits until the job specified as the first argument is finished, but it can be used in
-more custom ways by providing additional args including your own callback with custom code to be called after each
-status poll.
+By default, like in the most basic usage above in :ref:`flare_api_implementation_notes`, ``monitor_job()`` waits until
+the job specified as the first argument is finished, but it can be used in more custom ways by providing additional args
+including your own callback with custom code to be called after each status poll. The following is the API spec for
+monitor_job:
+
+.. code:: python
+
+    def monitor_job(
+        self, job_id: str, timeout: int = 0, poll_interval: float = 2.0, cb=None, *cb_args, **cb_kwargs
+    ) -> MonitorReturnCode:
+        """Monitor the job progress until one of the conditions occurs:
+         - job is done
+         - timeout
+         - the status_cb returns False
+
+        Args:
+            job_id: the job to be monitored
+            timeout: how long to monitor. If 0, never time out.
+            poll_interval: how often to poll job status
+            cb: if provided, callback to be called after each poll
+
+        Returns: a MonitorReturnCode
+
+        Every time the cb is called, it must return a bool indicating whether the monitor
+        should continue. If False, this method ends.
+
+        """
+
+Only the first argument is required, but with additional args, you can customize ``monitor_job()`` to do almost
+anything you want to do. The following is an example from the Jupyter notebooks for the hello-examples where you
+can see the usage of a sample_cb and cb_kwargs. This callback always returns True, keeping the default behavior of
+``monitor_job()`` of waiting until the job specified as the first argument is finished, but you can customize this to
+behave as you want.
+
+.. code:: python
+
+    def sample_cb(
```

Comment:
I will be moving a simple callback into flare_api as an example to use, but I don't know about having a default callback in use since the default of no callback seems more appropriate for the default behavior.

This will be in a future commit though.

Reply:
This output here prints a little more information, but I have some changes in another branch that may conflict with this one if I make further changes here, so I can make those changes in the other branch.